stdlib/public/core/UnicodeScalarProperties.swift

```diff
@@ -782,11 +784,29 @@ extension Unicode.Scalar.Properties {
       let scalar = Unicode.Scalar(
         _unchecked: UInt32(Int(_scalar.value) &+ Int(mappingDistance))
       )
-      return String(scalar)
+
+      return (scalar: scalar, specialMapping: nil)
     }
 
     // We did not have any mapping. Return the scalar as is.
-    return String(_scalar)
+    return (scalar: _scalar, specialMapping: nil)
+  }
+
+  fileprivate func _mappingToString(
+    _ mapping: (Unicode.Scalar?, UnsafeBufferPointer<UInt8>?)
+  ) -> String {
+    switch _getMapping(.lowercase) {
```

Comment:
If I understand correctly this should use the parameter mapping, right? :)
```suggestion
    switch mapping {
```